libafl_qemu/librasan/asan/src/allocator/backend/mod.rs

```diff
@@ -0,0 +1,58 @@
+//! # backend
+//! The backend is responsible for allocating the underlying memory used by the
+//! application. At present there is only one implemented allocator:
+//!
+//! - `dlmalloc` - A pure rust allocator based on the `dlmalloc` crate.
+//!
+//! A number other of possible implementations could be considered:
+//! - A simple bump allocator allocating from a fixed memory buffer
+//! - An allocator which calls down into the original `libc` implementation of `malloc`
+
+use alloc::fmt::Debug;
+use core::alloc::GlobalAlloc;
+
+use spin::Mutex;
+
+use crate::GuestAddr;
+
+#[cfg(feature = "dlmalloc")]
+pub mod dlmalloc;
+
+#[cfg(feature = "mimalloc")]
+pub mod mimalloc;
+
+pub trait AllocatorBackend: Sized + Debug + Send {
```

Comment:
Can't we directly use the `GlobalAlloc` trait here, and be compatible with other allocators?

Reply:
Done. It actually makes the code a bit cleaner (except for mocking in the tests). But when integrating a new allocator care needs to be taken as to its dependencies. e.g. if the allocator calls `malloc` (e.g. to store metadata, or as a result of some dependency then hideous recursion will occur).